File: Zarina/TaZQL/src/ast-tmp/expression/variables/Id.java
package ast.expression.variables;

import ast.expression.Expression;
import ast.expression.IExpressionVisitor;


public class Id extends Expression {

	private String id;
	
	public Id(String id) {
		this.id = id;
	}
	
	public String getID() {
		return id;
	}

	@Override
	public String toString() {
		return id;
	}

	 @Override
	 public <T> T accept(IExpressionVisitor<T> visitor) {
		 return visitor.visit(this);
	 }
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expressions/Id.cs
using AST.Nodes.Interfaces;

namespace AST.Nodes.Expressions
{
    public class Id : Expression, IHasType
    {
        public string Name { get; private set; }
        private Types.Type type = new Types.UndefinedType();
        
        public Id(string name, PositionInText position)
            : base(position)
        {
            this.Name = name;
        }

        public override T Accept<T>(VisitorInterfaces.IExpressionVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public Types.Type RetrieveType()
        {
            return this.type;
        }

        public void SetType(Types.Type type)
        {
            this.type = type;
        }

        public override bool Equals(object obj)
        {
            return (this.Name.Equals(((Id)obj).Name));
        }

        public override int GetHashCode()
        {
            return this.Name.GetHashCode();
        }
    }
}
